apps/data/src/tasks/data_upload_task.py

```
@@ -0,0 +1,415 @@
+        write_upload_metadata("failed", None, str(e))
+        return {
+            "status": "error",
+            "error_message": str(e),
+            "experiment_id": EXPERIMENT_ID,
+            "source_kind": SOURCE_KIND,
+            "upload_table_name": UPLOAD_TABLE_NAME,
+            "upload_id": UPLOAD_ID,
+        }
+
+result = main()
+logger.info(f"Status: {result['status']}")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)